gap/utils.gi

```diff
@@ -577,3 +577,27 @@ function(nr, n)
   od;
   return x;
 end);
+
+InstallGlobalFunction(DError,
```

Comment:
I'm happy for this to be introduced, although I have a reservation that I'll detail in a minute. But at the very least I think the introduction of this function should be mentioned in the commit message; preferably in a separate commit; and ideally in a separate PR (although no big deal on that front).

The potential pitfall is about what happens when the first variable that you introduce after the intended error message is itself a string:
```
gap> variable := "wilf";;
gap> DError("Hi {} you've done something wrong!", variable);
Error, out of bounds -- used 1 replacement fields without id when there are only 0 arguments at /Users/Wilf/gap/lib/string.gi:1353 called from
CallFuncList( PrintToFormatted,
 Concatenation( [ OutputTextString( str, false ), s ], data )
 ); at /Users/Wilf/gap/lib/string.gi:1395 called from
CallFuncList( StringFormatted, arg{[ i - 1 .. Length( arg ) ]}
 ) at /Users/Wilf/gap/pkg/digraphs/gap/utils.gi:592 called from
<function "DError">( <arguments> )
```
Note also:
```
gap> DError("Hi {} you've done something wrong!", variable, fail);
Error, Hi fail you've done something wrong!wilf called from
not in any function at *stdin*:2
type 'quit;' to quit to outer loop
```
`StringFormatted` itself doesn't suffer from this problem because it requires the main error message string to be the first argument.

I like your idea of introducing this function, and of allowing the error message to be broken into multiple smaller strings. But if we choose this option, then it means that we need to be careful. Here are some options I see.

1. Be careful never to use a string for the first replacement argument (sounds too risky and having a variable be a string is realistic use case in my opinion).
2. We introduce a dummy separator argument to mark the boundary between message and replacement (at least required in cases where the first variable _could_ be a string), e.g.
```
DError(msg1, msg2, msg3, DIGRAPHS_SEPARATOR, var1, var2)
```
3. We 'give up' and use `Concatenation` around the message arguments so that they all fit into the first argument (uglier but works):
```
DError(Concatenation(msg1, msg2, msg3), var1, var2)
```
4. We 'give up' and fit everything into the first argument, but we start using multiline strings (would require some additional processing; see this example where it doesn't look quite right; might also require changes to the linter):
```
gap> DError("""this is the start of a very long error message and
> I am going to break it over multiple
> ({}) lines because I am cool""", 3);
Error, this is the start of a very long error message andI am going to break it over \
multiple(3) lines because I am cool called from
not in any function at *stdin*:36
```
Any other ideas?

Reply:
This is a good point, I also wondered about strings as variables, and somehow convinced myself that it shouldn't be an issue. I think you've convinced me the other way around now :) 

I'm happy to introduce this in a separate PR too, no problem. 

About what to do to resolve the issues of strings as variables: 

1. I agree we shouldn't do this
2+3. We could do something between 2 and 3: `DError([msg1, msg2, msg3], var1, var2);` minimally different, and avoids using the too long `Concatenation`? Alternatively, we could consume msg1, msg2,.. until we hit a non-string or we hit `DIGRAPHS_SEPARATOR`, thereby allowing `DError` to be used without `DIGRAPHS_SEPARATOR` if none of the variables is a string, and use it if the first variable is a string. 
4. I'm not enamoured with this option mostly because the resulting strings will be badly formatted in the code.

Reply:
I'm happy to not do 1 and 4 and I like both of your suggestions for something between 2 and 3. I think my preference is for the list version, which would permit these variations:
```
DError([msg1, msg2, msg3], var1, var2);
DError([msg1], var1, var2);
DError(msg1, var1, var2);
```
which adds minimal ugliness.